server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -865,7 +865,9 @@ public void executeFetchPhase(ShardFetchRequest request, SearchShardTask task, A
                     }
                     executor.success();
                 }
-                return searchContext.fetchResult();
+                var fetchResult = searchContext.fetchResult();
+                fetchResult.incRef();
```

Comment:
This is a nasty pattern admittedly but we do the same (instantiate on the context -> inc ref -> return) for the query results so I went for the same approach here as well for now.

Reply:
Agree, nasty but not a new pattern. Can we resolved as a followup.